JASP-Engine/JASP/R/descriptives.R

```diff
@@ -1317,4 +1329,69 @@ Descriptives <- function(jaspResults, dataset, options) {
 
   return(plotObj)
 }
+
+.descriptivesScatterPlots <- function(jaspContainer, dataset, variables, split, options) {
+
+  if (!is.null(split) && split != "") {
+    group <- dataset[, .v(split)]
+    legendTitle <- split
+    if (options[["showLegend"]]) {
+      oldPalette <- JASPgraphs::getGraphOption("palette")
+      on.exit(JASPgraphs::setGraphOption("palette", oldPalette))
+      JASPgraphs::setGraphOption("palette", options[["colorPalette"]])
```

Comment:
This seems kinda error prone. People will forget to do this.
Can't we pass this in as a parameter to graphing functions if we don't want to make persistent changes to JASPgraphs' state?

Reply:
And here we can move it to common